### ISSUE TITLE
mounts.path containing ~ (tilde) results in gopass git ls-files fails; expansion needed

### ISSUE THREAD

**Issue by quite** [gopasspw/gopass#2982]
### Summary
If `mounts.path` contains a path beginning with tilde, then `gopass git ls-files` fails because it gets passed the tilde without it being first expanded.

### Steps To Reproduce
If `.config/gopass/config` contains:
```
[mounts]
	path = ~/.password-store
```

Then `gopass git ls-files` says:

```
⚠ Running 'git ls-files' in ~/.password-store...
2024/10/29 15:33:36 chdir ~/.password-store: no such file or directory
```

### Expected behavior

The tilde needs to be expanded before `git ls-files` is fed with the path.

Note that there may be other commands which is fed with the raw path, but where the tilde needs to be expanded first.

### Environment

- OS: Arch Linux
- OS version: Linux kiwi 6.10.5-arch1-1 # 1 SMP PREEMPT_DYNAMIC Thu, 15 Aug 2024 00:25:30 +0000 x86_64 GNU/Linux
- gopass Version: 1.15.14
- Installation method: Arch Linux package

**Comment by Denis-Kuso** [comment]
Hi. Is any help required for this issue? If so, can I try?

**Comment by quite** [comment]
One need to dig into the code for the `gopass git` subcommand, and make sure that the store path that it uses gets tilde-expanded. And there may be other subcommands which needs this treatment. Please help out if you can, I don't have time.

**Comment by Denis-Kuso** [comment]
No problem, I'll have a go. Can you please assign me to this issue? Thank you.

**Comment by quite** [comment]
No, I don't have permission to do that. I suggest you just work on a solution and submit a patch. :green_heart:

**Comment by Denis-Kuso** [comment]
After some testing, I was able to reproduce the issue. I think the issue
was resolved by the commit #2961 (part of version upgrade to 1.15.15) and hence
no longer occurs in the newest release.

I used separate Dockerfiles (example below) with different gopass versions (1.15.14 and 1.15.15)
and different OS distributions (Arch and Ubuntu) to account for author's env.

I've attached a screenshot showing the versions, `mounts.path` value, existence
of that directory and output of `gopass git ls-files`.
It demonstrates the behavior I encountered while testing, i.e. issue no longer occurs
in 1.15.15,but occurs in 1.15.14 independent of OS.

![Screenshot 2024-12-10 at 19 07 12](https://github.com/user-attachments/assets/0f9b4cf0-9e5e-4ca3-8169-7b334f2461f5)


Let me know if you would like further details on anything and/or how to proceed further.

<details>

<summary>example Dockerfile</summary>

```
FROM archlinux:latest

# Update system and install necessary packages
RUN pacman -Syu --noconfirm \
    && pacman -S --noconfirm \
        curl \
        tar \
        bash \
        vim \
        git \
        gnupg

# Download and extract the gopass binary
RUN curl -L https://github.com/gopasspw/gopass/releases/download/v1.15.14/gopass-1.15.14-linux-amd64.tar.gz -o /tmp/gopass.tar.gz \
    && tar -xvzf /tmp/gopass.tar.gz -C /usr/local/bin/ \
    && rm /tmp/gopass.tar.gz

CMD ["/bin/bash"]
```

</details>

**Comment by quite** [comment]
Great, it's fixed! Thank you
